Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift

```diff
@@ -70,36 +71,23 @@ extension LogtoWebViewAuthViewController: WKScriptMessageHandler {
     public func userContentController(_: WKUserContentController, didReceive message: WKScriptMessage) {
         guard let body = SocialPostBody.safeParseJson(json: message.body),
               let redirectUri = URL(string: body.redirectTo),
-              let callbackUri = URL(string: body.callbackUri),
-              var callbackComponents = URLComponents(url: callbackUri, resolvingAgainstBaseURL: true)
+              let redirectScheme = URLComponents(url: redirectUri, resolvingAgainstBaseURL: true)?.scheme,
+              let callbackUri = URL(string: body.callbackUri)
         else {
             return
         }
 
-        let session = ASWebAuthenticationSession(
-            url: redirectUri,
-            callbackURLScheme: LogtoWebViewAuthViewController.webAuthCallbackScheme
-        ) { [self] customUri, error in
-            guard let customUri = customUri else {
-                postErrorMessage(.webAuthFailed(innerError: error))
-                return
-            }
-
-            // Construct callback URL for WebView
-            let customComponents = URLComponents(url: customUri, resolvingAgainstBaseURL: true)
-            callbackComponents
-                .queryItems = (callbackComponents.queryItems ?? []) + (customComponents?.queryItems ?? [])
-
-            guard let url = callbackComponents.url else {
-                postErrorMessage(.unableToConstructCallbackUri)
-                return
-            }
-
-            self.webView.load(URLRequest(url: url))
+        guard let socialPlugin = authSession.socialPlugins.first(where: {
+            $0.urlSchemes.contains(redirectScheme)
+        }) else {
+            // TO-DO: error handling
+            return
         }
 
-        session.prefersEphemeralWebBrowserSession = true
-        session.presentationContextProvider = self
-        session.start()
+        socialPlugin
+            .start(LogtoSocialPluginConfiguration(redirectUri: redirectUri, callbackUri: callbackUri,
```

Comment:
nit: not sure about the swift practice, will it be better to define a LogtoSocialPluginConfiguration variable here.

Reply:
i think it's okay for now - let's define the var when it has more params